src/crafting.h

```diff
@@ -52,6 +52,7 @@ struct recipe {
     std::map<skill_id, int> required_skills;
     bool reversible; // can the item be disassembled?
     bool autolearn; // do we learn it just by leveling skills?
+    bool uncraft_single_charge; // recipe is to uncraft one charge of an item with multiple default charges
```

Comment:
This member is not initialized anywhere, reading from it invokes undefined behavior. You need to load it from JSON, probably in a similar way as the other booleans (above) are loaded.